src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -24,6 +25,19 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node,
                 ? node.Parent.Parent as ExpressionSyntax
                 : node.Parent as ExpressionSyntax;
 
+            // Have to be careful if we would remove parens and cause a + and a + to become a ++.
+            // (same with - as well).
+            var tokenBeforeParen = node.GetFirstToken().GetPreviousToken();
```

Comment:
Does this have problems in script/interactive mode?

Reply:
Fortunately no.  `GetPreviousToken` will simply return `default(SyntaxToken)`.  `default(SyntaxToken)` returns empty-string for `.Text`, and `LastOrDefault` will just return `0` in that case.